shaders/src/main.fs

```diff
@@ -28,6 +28,17 @@ void main() {
 
     fragColor = evaluateMaterial(inputs);
 
+#if defined(HAS_FOG)
+    // we need the real world position, which is obtained by adding the world-offset
+    vec3 wsCameraPostion = getWorldCameraPosition() + getWorldOffset();
+    vec3 wsFragPosition = getWorldPosition() + getWorldOffset();
+
+    // now we need the world-space vector from camera to the shaded fragment
+    vec3 cameraToWorld = wsFragPosition - wsCameraPostion;
```

Comment:
`cameraToWorld` sounds like the name of a matrix. Technically it's the view vector `v`.

Reply:
oh right, actually, I should be able to use `shading_view`

Reply:
Actually shading_view is normalized, which wouldn't work here. Also it's the opposite vector we need here (i.e. -shading_view_not_normalized).